modules/OtJsonUtilities.js

```diff
@@ -85,24 +85,26 @@ class OtJsonUtilities {
             datasetCopy = Utilities.copyObject(dataset);
             OtJsonUtilities.sortGraphRelationsAndIdentifiers(datasetCopy['@graph']);
             return datasetCopy;
+        case '1.1':
+            return undefined;
         default:
             throw new Error('Unsupported ot-json version!');
         }
     }
 
-    /**
-     * Formats the dataset IN PLACE so that the signature can be generated properly
-     *
-     * @param dataset
-     * @returns {any}|undefined
-     */
-    static prepareDatasetForExtractSigner(dataset) {
+    static prepareDatasetForDatabaseRead(dataset) {
         const version = OtJsonUtilities._getDatasetVersion(dataset);
 
+        let datasetCopy;
+
         switch (version) {
         case '1.0':
-            OtJsonUtilities.sortGraphRelationsAndIdentifiers(dataset['@graph']);
-            break;
```

Comment:
Didn't we have an issue here about having to sort in place due to side effects? How did we overcome this issue?

Reply:
Yes, but it seems that the side effect doesn't change normal functioning of node. Only one unit test was affected by this side effect (Convert XMLs into OT-JSON and vice versa).